code/components/citizen-server-impl/src/NoticeLogicProcessor.cpp

```diff
@@ -0,0 +1,140 @@
+#include <StdInc.h>
+#include "NoticeLogicProcessor.h"
+
+NoticeLogicProcessor::NoticeLogicProcessor(fx::ServerInstanceBase* server)
+{
+	m_cvManager = server->GetComponent<console::Context>()->GetVariableManager();
+	m_resManager = server->GetComponent<fx::ResourceManager>()->GetCurrent();
+}
+
+bool NoticeLogicProcessor::ProcessNoticeRule(json& ruleRef, uint32_t nestingLevel)
+{
+	if (nestingLevel >= 10)
+		throw std::invalid_argument("Maximum nesting level for notice rules was exceeded");
```

Comment:
We need to catch this somewhere

Reply:
NoticeLogicProcessor exceptions are caught by its consumer, [DownloadAndProcessNotices in ServerNucleus.cpp](https://github.com/jakub-cfx/fivem/blob/tweak/server-notice-logic/code/components/citizen-server-impl/src/ServerNucleus.cpp#L35)
I could move the try-catch from there to BeginProcessingNotices so its contained within NoticeLogicProcessor but I feel like the handling of exceptions should be left to the highest level consumer

Reply:
Looks good, as this is no game or performance heavy code I think it's fine to keep it like this.